src/cms/page_xliff_converter.py

```diff
@@ -1,358 +1,274 @@
+
+        :return: pretty printed XML source code
+        :rtype: str
+        """
+        return etree.tostring(self.xliff, pretty_print=True)
+
+
+class PageXliffHelper():
+    """
+    Wrapper and helper methods for reading/writing XLIFF files and saving PageTranslation
+    """
+    def __init__(self, src_lang=None, tgt_lang=None):
+        """
+        :param src_lang: source language of tran
+        :param src_lang: source language of translation
+        :type src_lang: cms.models.languages.language.Language
+        :param tgt_lang: target language of translation
+        :type tgt_lang: cms.models.languages.language.Language
+        """
+        if src_lang is not None:
+            self.src_lang = src_lang
+        if tgt_lang is not None:
+            self.tgt_lang = tgt_lang
 
     @staticmethod
     def save_file(content, file_path):
+        """
+        Write text to file
+
+        :param content: content of file
+        :type content: str
+        :param file_path: path to newly created file
+        :type file_path: str
+        """
         os.makedirs(os.path.dirname(file_path), exist_ok=True)
         with open(file_path, 'w', encoding='utf-8') as file:
             file.write(content)
 
-    def export_page_translation_xliff(self, source_translation_page, target_language_code, filename=None):
+    def export_page_translation_xliff(self, page):
+        """
+        Create XLIFF file for source page translation and target language
+
+        :param page: page of translation that should be exported
+        :type page: cms.models.pages.page.Page
+        :return: file path to XLIFF file
+        :rtype: string or None
+        """
+        converter = TranslationXliffConverter(self.src_lang, self.tgt_lang)
+
+        src_trans = PageTranslation.objects.get(page=page, language=self.src_lang)
+        tgt_trans = PageTranslation.objects.filter(page=page, language=self.tgt_lang)
+        if len(tgt_trans) == 1:
+            tgt_trans = tgt_trans[0]
+        else:
+            tgt_trans = PageTranslation.objects.create(
+                title="",
+                text="",
+                status=src_trans.status,
+                language=self.tgt_lang,
+                page=page)
+
+        region_slug = page.region.slug
+        slug = src_trans.slug
 
-        if not filename:
-            filename = f'page_{source_translation_page.page.id}_{source_translation_page.language.code}_{target_language_code}.xliff'
+        # properties that make a translation unique: page id, target language, source version
+        xliff_id = self.tgt_lang.code + "_" + str(page.id) + "_" + str(src_trans.version)
 
-        file_path = None
+        filename = f"{region_slug}_{self.src_lang.code}__{xliff_id}__{slug}.xliff"
 
-        xliff_content = self.converter.page_translation_to_xliff(source_translation_page, target_language_code)
+        xliff_content = converter.translation_to_xliff(page.id, src_trans, tgt_trans, xliff_id)
```

Comment:
I see two problems with this:

1. If the source language doesn't exist for one page, this still gives me a `DoesNotExist`-error
2. If an empty target translation is being created, it has the side-effect that the new target translation is marked with a :heavy_check_mark: because it is more recent than the source translation. So I would suggest to either allow the conversion functions to work with empty default values or to delete the `tgt_trans` object after it has been converted to xliff.

Reply:
Apart from that, you could replace the conditional blocks with [get_or_create()](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#get-or-create)

Reply:
1. should be addressed now with a `filter()` instead of `get()`, while checking if there is exactly one result.
2. Even if the `tgt_trans` is not being saved?

Reply:
1. Yes this looks good :+1: 
2. Yes, but you could use the constructor `PageTranslation()` instead of `PageTranslation.objects.create()` to instantiate a translation object which is not saved if you don't call `save()`.